components/dispatcher/internal/dispatcher/dispatcher.go

```diff
@@ -139,11 +143,12 @@ func (d *Dispatcher) handleKafkaMessages(consumer kafkaconsumer.ConsumerInterfac
 				zap.String("Topic", *e.TopicPartition.Topic),
 				zap.Int32("Partition", e.TopicPartition.Partition),
 				zap.Any("Offset", e.TopicPartition.Offset))
-			_ = d.Client.Dispatch(e.Value) // Ignore Errors - Dispatcher Will Retry And We're Moving On!
+
+			_ = d.Client.Dispatch(convertToCloudEvent(e)) // Ignore Errors - Dispatcher Will Retry And We're Moving On!
```

Comment:
Why convert this to a cloud event on the outbound side? Shouldn't that be done in the channel?

Reply:
Channel writes kafka messages to kafka... dispatcher reads kafka messages from kafka and sends them out as cloudevents

Reply:
Are they stored in kafka differently? I recall seeing the "KafkaHeaders" being set.

Reply:
It's clear now. Thanks for filling me in 😄